packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.styles.ts

```diff
@@ -0,0 +1,127 @@
+
+export const getStyles = memoizeFunction((
+  theme: ITheme,
+  customStyles?: ICheckboxStyles
+): ICheckboxStyles => {
+  const { semanticColors, palette } = theme;
+  const checkmarkFontColor = semanticColors.inputForegroundChecked;
+  const checkboxBorderColor = semanticColors.inputBorder;
+  const checkboxBorderHoveredColor = semanticColors.inputBorderHovered;
+  const checkboxBackgroundChecked = semanticColors.inputBackgroundChecked;
+  const checkboxBackgroundCheckedHovered = semanticColors.inputBackgroundCheckedHovered;
+  const checkboxBackgroundDisabled = semanticColors.disabledText;
+  const checkboxTextColorDisabled = semanticColors.disabledText;
+
+  const styles: ICheckboxStyles = {
+    root: [
+      getFocusStyle(theme),
+      {
+        padding: '0',
+        border: 'none',
+        background: 'none',
+        margin: '0',
+        outline: 'none',
+        display: 'block',
+        cursor: 'pointer',
+
+        // This is to hide the label's overflow because of the negative margin.
+        overflow: 'hidden'
+      }
+    ],
+    label: {
+      display: 'inline-flex',
+      alignItems: 'center',
+      cursor: 'pointer',
+
+      // The negative margin is in place to correct for the margin of the box
+      margin: '0 -4px',
+      position: 'relative',
+      userSelect: 'none'
+    },
+    labelReversed: {
+      flexDirection: 'row-reverse',
+      justifyContent: 'flex-end'
+    },
+    labelDisabled: {
+      cursor: 'default'
+    },
+    input: {
+      position: 'absolute',
+      opacity: '0'
+    },
+    box: {
+      display: 'flex',
+      margin: '0 4px',
+      alignItems: 'center',
+      justifyContent: 'center',
+      flex: '0 0 ' + MS_CHECKBOX_LABEL_SIZE,
+      height: MS_CHECKBOX_LABEL_SIZE,
+    },
+    checkbox: {
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      height: MS_CHECKBOX_LABEL_SIZE,
+      width: MS_CHECKBOX_LABEL_SIZE,
+      borderWidth: '1px',
+      borderStyle: 'solid',
+      borderColor: checkboxBorderColor,
+      boxSizing: 'border-box',
+      transitionProperty: 'background, border, border-color',
+      transitionDuration: MS_CHECKBOX_TRANSITION_DURATION,
+      transitionTimingFunction: MS_CHECKBOX_TRANSITION_TIMING,
+
+      /* incase the icon is bigger than the box */
+      overflow: 'hidden',
+    },
+    checkboxHovered: {
+      borderColor: checkboxBorderHoveredColor,
+    },
+    checkboxChecked: {
+      background: checkboxBackgroundChecked,
+    },
+    checkboxCheckedHovered: {
+      background: checkboxBackgroundCheckedHovered,
+    },
+    checkboxDisabled: {
+      background: checkboxBackgroundDisabled,
+    },
+    checkboxCheckedDisabled: {
+      background: checkboxBackgroundDisabled,
+    },
+    checkmark: {
+      opacity: '0',
+      color: checkmarkFontColor
+    },
+    checkmarkChecked: {
+      opacity: '1'
+    },
+    checkmarkDisabled: {
+    },
+    checkmarkCheckedDisabled: {

```

Comment:
We're moving away from fabric palette colors to semantic color slots. We're asking people do that in conjunction with glamorizing components because they're looking at the styles anyways, saves them time from coming back and doing it later. They're already learning something new that's related, and it's easier to do while they have in mind how that component is structuring its styles. This also saves us having to do a second full pass of all our components.